src/ParallelAlgorithms/Interpolation/Callbacks/ObserveTimeSeriesOnSurface.hpp

```diff
@@ -41,8 +41,24 @@ namespace callbacks {
 namespace detail {
 
 template <typename... Ts>
-auto make_legend(tmpl::list<Ts...> /* meta */) {
-  return std::vector<std::string>{"Time", db::tag_name<Ts>()...};
+auto make_legend(tmpl::list<Ts...>) {
+    std::vector<std::string> legend = {"Time"};
+
+    // Helper lambda to append tag names and print
+    auto append_and_print = [&legend](auto tag_name) {
+        std::string name = db::tag_name<decltype(tag_name)>();
+        if (name == "DimensionfulSpinVector") {
```

Comment:
Don't hardcode this to "DimensionalfulSpinVector" please. Can you perhaps check if the type of the tag is a sequence and if so, expand the components?